src/rpdk/core/contract/resource_client.py

```diff
@@ -377,6 +376,10 @@ def call(self, action, request):
             response = self._call(payload)
             status = OperationStatus[response["status"]]
 
+        # ensure writeOnlyProperties are not returned on final responses
+        if "resourceModel" in response.keys():
```

Comment:
Do we need this check? Would there be a time when the response will not have "resourceModel"?

Reply:
Some of the tests don't set it up, and it's possible that a model is not returned, e.g; on a `Delete` action being successful?